server/config/config.go

```diff
@@ -315,7 +322,31 @@ func (c *ServerConfig) ShouldDiscover() bool { return c.DiscoveryURL != "" }
 func (c *ServerConfig) ReqTimeout() time.Duration {
 	// 5s for queue waiting, computation and disk IO delay
 	// + 2 * election timeout for possible leader election
-	return 5*time.Second + 2*time.Duration(c.ElectionTicks*int(c.TickMs))*time.Millisecond
+	// return 5*time.Second + 2*time.Duration(c.ElectionTicks*int(c.TickMs))*time.Millisecond
+
+	// OCP PATCH: use an environment variable to configure the non election-related timeout
+	// value used. If it isn't configured, default to 25 seconds.
+	// Default of 25 seconds gets us to a total of 27s on "standard hardware" and 30s on "slow hardware"
+	// due to the configurations of election timeouts in https://github.com/openshift/cluster-etcd-operator/blob/5bbe49442101475febb89bba3be808aa121f5c0c/pkg/hwspeedhelpers/hwhelper.go#L21-L33
+    // This gets us closer to the kube-apiserver's upper bound request timeout of 34 seconds while leaving some time
+    // for other things to have on the kube-apiserver side of things.
+    // See: https://issues.redhat.com/browse/OCPBUGS-50510
+    // TODO(everettraven): pursue an upstream fix for this so we don't have to carry this patch for forever
```

Comment:
nit: inconsistent (space vs. tab) indent.  Looks like you aren't currently running a `go fmt ...` pre-check?  See [here][1] if you want something you might be able to copy/paste over.

[1]: https://github.com/openshift/release/blob/855ee03a5956295152ebe41d166ef344d3729ae5/ci-operator/config/openshift/cluster-version-operator/openshift-cluster-version-operator-main.yaml#L54-L58